src/Sarif/Core/ExceptionData.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    /// <summary>
+    /// Describes a condition relevant to the tool itself, as opposed to being relevant to a file being analyzed by the tool.
+    /// </summary>
+    public partial class ExceptionData
+    {
+        public static ExceptionData Create(Exception exception)
+        {
+            return new ExceptionData
+            {
+                Type = exception.GetType().Name,
+                Message = exception.Message,
+                InnerExceptions = GetInnerExceptions(exception),
+                Stack = Stack.Create(exception.StackTrace)
+            };
+        }
+
+        private static IList<ExceptionData> GetInnerExceptions(Exception exception)
+        {
+            var innerExceptions = new List<ExceptionData>();
+
+            IReadOnlyCollection<Exception> aggregateInnerExceptions = (exception as AggregateException)?.InnerExceptions;
+            if (aggregateInnerExceptions != null)
+            {
+                foreach (Exception innerException in aggregateInnerExceptions)
```

Comment:
Nice, I didn't know about AggregatedException.Flatten. But now that I know about it... I'm not sure I like its semantics. I think the causal chain for each exception in InnerExceptions is nice to have, and the code already does it. I'd like to leave as is.

---
In reply to: [61638878](https://github.com/Microsoft/sarif-sdk/pull/229#discussion_r61638878) [](ancestors = 61638878)